lib/Interpreter/CppInterOp.cpp

```diff
@@ -3082,23 +3082,26 @@ namespace Cpp {
     clang::DeclContext *DC;
     clang::DeclContext::decl_iterator decl;
 
+    llvm::SmallVector<clang::DeclContext*, 4> DCs;
+
     if (auto *TD = dyn_cast_or_null<TagDecl>(D)) {
       DC = clang::TagDecl::castToDeclContext(TD);
-      decl = DC->decls_begin();
-      decl++;
+      DCs.push_back(DC);
     } else if (auto *ND = dyn_cast_or_null<NamespaceDecl>(D)) {
       DC = clang::NamespaceDecl::castToDeclContext(ND);
-      decl = DC->decls_begin();
+      DC->collectAllContexts(DCs);
     } else if (auto *TUD = dyn_cast_or_null<TranslationUnitDecl>(D)) {
       DC = clang::TranslationUnitDecl::castToDeclContext(TUD);
-      decl = DC->decls_begin();
+      DC->collectAllContexts(DCs);
     } else {
       return;
     }
```

Comment:
We need something:

if(!isa<DeclContext>(D))
  return;
DC = cast<DeclContext>(D);
DC->collect(...);
...

Reply:
It may not be required.
We already check for [`TagDecl`](https://clang.llvm.org/doxygen/classclang_1_1TagDecl.html), [`NamespaceDecl`](https://clang.llvm.org/doxygen/classclang_1_1NamespaceDecl.html), & [`TranslationUnitDecl`](https://clang.llvm.org/doxygen/classclang_1_1TranslationUnitDecl.html).
And looking at the documentation for them, they inherited `DeclContext`.

Let me know, if you still think it is necessery.

Reply:
I have fixed it. Please have a look.

Reply:
Looks good but I am not seeing my fixme comment addressed.

Reply:
Done. Sorry, I missed it earlier.